pgjdbc/src/main/java/com/yugabyte/ysql/LoadBalanceManager.java

```diff
@@ -75,10 +75,7 @@ public static boolean needsRefresh(long refreshInterval) {
     return false;
   }
 
-  public static synchronized boolean refresh(Connection conn, long refreshInterval) throws SQLException {
-    if (!needsRefresh(refreshInterval)) {
```

Comment:
`needsRefresh()` is always called before `refresh()` in a `synchronized` block now. Not needed here any longer.